src/NeoSharp.Core/DI/Modules/NetworkModule.cs

```diff
@@ -36,9 +35,22 @@ public void Register(IContainerBuilder containerBuilder)
                             t != typeof(MessageHandlerProxy))
                 .ToArray();
 
-            containerBuilder.Register(typeof(IMessageHandler<>), messageHandlerTypes);
-            containerBuilder.RegisterInstanceCreator<IMessageHandler<Message>>(c =>
-                new MessageHandlerProxy(c, messageHandlerTypes, c.Resolve<ILogger<MessageHandlerProxy>>()));
+            containerBuilder.RegisterCollection(typeof(IMessageHandler<>), messageHandlerTypes);
```

Comment:
What's the point in double registration?

Reply:
For sure is not need... maybe was because of the MessageHandlerProxy registration that I left there.

this class will have his own interface and don't need to share the IMessageHandler<T> or even the new IMessageHandler.